### PR TITLE
Move VolumeDriver to HostConfig

### PR DESCRIPTION
**Pull request by jimmyxian**
Depend on : https://github.com/samalba/dockerclient/pull/182

Fix: https://github.com/docker/swarm/issues/1339
Maybe fix this issue https://github.com/docker/swarm/issues/1077 as well :)

Signed-off-by: Xian Chaobo <xianchaobo@huawei.com>

### CODE REVIEW

**Comment by jimmyxian** [comment]
ping @docker/swarm-maintainers

**Comment by abronan** [comment]
LGTM thanks @jimmyxian :)

File: cluster/swarm/cluster.go [hunk 1]
```diff
@@ -115,11 +116,14 @@ func (c *Cluster) CreateContainer(config *cluster.ContainerConfig, name string)
 	container, err := c.createContainer(config, name, false)
 
 	//  fails with image not found, then try to reschedule with soft-image-affinity
-	if err != nil && strings.HasSuffix(err.Error(), "not found") && !config.HaveNodeConstraint() {
-		// Check if the image exists in the cluster
-		// If exists, retry with a soft-image-affinity
-		if image := c.Image(config.Image); image != nil {
-			container, err = c.createContainer(config, name, true)
+	if err != nil {
+		bImageNotFoundError, _ := regexp.MatchString("image .* not found", err.Error())
```
Comment (dongluochen) [inline cluster/swarm/cluster.go]: nit: '.*' might be too general. Is '\S*' better?

Comment (jimmyxian) [inline cluster/swarm/cluster.go]: OK

**Comment by dongluochen** [comment]
LGTM. Thanks @jimmyxian!

**Comment by abronan** [comment]
ping @vieux

**Comment by vieux** [comment]
LGTM

**Pull request merged**